rfcs/text/formik.md

```diff
@@ -0,0 +1,399 @@
+documentation on [fully-fledged form solutions](https://reactjs.org/docs/forms.html#fully-fledged-solutions) mentions:
+
+> If you're looking for a complete solution including validation, keeping track of the visited fields, and handling
+> form submission, [Formik](https://jaredpalmer.com/formik) is one of the popular choices.
+
+The purpose of this RFC is to discuss the advantages and disadvantages of deprecating Carbon's existing form and
+validation functionality and integrating [Formik](https://jaredpalmer.com/formik) as a replacement.
+
+There's plenty of general discussion online about "reinventing the wheel" (see
+[Stack Exchange](https://softwareengineering.stackexchange.com/questions/29513/is-reinventing-the-wheel-really-all-that-bad),
+[Quora](https://www.quora.com/Why-is-it-discouraged-to-reinvent-the-wheel-in-programming-Wont-it-give-me-a-deeper-and-better-understanding-of-how-things-work-underneath),
+[Jeff Atwood](https://blog.codinghorror.com/dont-reinvent-the-wheel-unless-you-plan-on-learning-more-about-wheels/),
+etc).  Some of the key advantages of "not reinventing the wheel" (ie: not building a custom solution, but using an
+existing third-party library) are:
+
+* **Less code to maintain**  –  Much time and effort can be saved by using a third-party library which is kept
+  well-maintained, thus freeing up one's development resources to focus on higher-priority project-unique functionality.
+* **Robustness**  –  A popular well-maintained library is likely to be thoroughly tested and debugged, especially for
+  edge cases which have been encountered and addressed by the library's community.
+* **Familiarity**  –  Users are potentially already familiar with popular third-party libraries.
+
+Considering the reasons why "reinventing the wheel" (ie: building a custom solution) can sometimes be beneficial in
+certain situations:
+
+* It can be a very educational experience for developers, but that isn't a relevant factor for this project.
+* A third-party library may be an unnecessarily heavy solution, if only a small percentage of its functionality will be
+  used by the application, and a custom-built solution may be much more lightweight.  But in this case, Formik is
+  [a relatively small library](https://jaredpalmer.com/formik/docs/overview)
+  ([14.7kB when minified and gzipped](https://bundlephobia.com/result?p=formik@2.0.4)), and we would be using a
+  significant percentage of its functionality.
+
+
+# Detailed design
+
+## Overview of Carbon's form and validation functionality
+
+Carbon provides the following components:
+
+* [**`<Form>`**](src/__experimental__/components/form/form.component.js)  –  This augments an HTML
+  [`<form>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) element with: a footer (containing "Save"
+  and "Cancel" buttons), form submission functionality, [CRSF](https://en.wikipedia.org/wiki/CRSF) functionality, a
```

Comment:
Typo CRSF -> CSRF

Reply:
Wow, well spotted, thanks! :clap: Fixed in db9e307